app/Http/Controllers/SponsorController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Redirect;
+use App\Sponsor;
+
+class SponsorController extends Controller
+{
+    public function index()
+    {
+        $sponsors = Sponsor::all();
+
+        return view('sponsors.index')->with([
+            'sponsors' => $sponsors,
+        ]);
+    }
+
+    public function create()
+    {
+        return view('sponsors.create');
+    }
+
+    public function store(Request $request)
+    {
+        $validated = $this->validate($request, [
+            'cnpj' => 'required|unique:sponsors',
+            'name' => 'nullable|string',
+            'email' => 'nullable|string'
+        ]);
+
+        Sponsor::create($validated);
+
+        return $this->index()->with([
+            'message_success' => "Patrocinador cadastrado com sucesso"
+        ]);
+    }
+
+    public function show($id)
+    {
+        $sponsor = Sponsor::find($id);
+
+        return view('sponsors.show')->with([
+            'sponsors' => $sponsor
+        ]);
+    }
+
+    public function edit($id)
+    {
+        $sponsor = Sponsor::find($id);
+
+        return view('sponsors.edit')->with([
+            'sponsors' => $sponsor
+        ]);
+    }
+
+    public function update($id, Request $request)
+    {
+        $sponsor = Sponsor::find($id);
+
+        $request->validate([
+            'cnpj' => 'required',
```

Comment:
Verificar que, diferentemente da inclusão, na alteração do registro está permitindo cadastrar dois patrocinadores com o mesmo CNPJ. Acredito que essa validação não foi aplicada na alteração pois o registro alterado estaria duplicado com o CNPJ dele mesmo. Nesse [artigo](https://www.csrhymes.com/2019/06/22/using-the-unique-validation-rule-in-laravel-form-request.html) é mostrada uma solução para esse caso.